### ISSUE TITLE
Tooltip behaviour not consistent with Vanilla

### ISSUE THREAD

**Issue by petermakowski** [canonical-web-and-design/react-components#697]
[Tooltip](https://canonical-web-and-design.github.io/react-components/?path=/story/tooltip--default-story) behaviour is not consistent with [Vanilla Tooltip](https://vanillaframework.io/docs/patterns/tooltips), e.g.:

- does not toggle on press or focus
- tooltip text is not announced by a screen reader
- (investigate only) tooltip often goes off screen on small breakpoint:
![image](https://user-images.githubusercontent.com/7452681/161027178-0f81c173-8604-4b4c-a5e6-3a159accbf8e.png)

- Tooltip does not support id prop and can't be used in combination with `aria-describedby` as suggested in vanilla docs: https://vanillaframework.io/docs/patterns/tooltips#accessibility
